src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -921,6 +921,18 @@ func (r *StatefulSetResource) CurrentVersion(ctx context.Context) (string, error
 	return stsVersion, nil
 }
 
+func (r *StatefulSetResource) IsManagedDecommission() (bool, error) {
+	t, ok := r.pandaCluster.GetAnnotations()[ManagedDecommissionAnnotation]
+	if !ok {
+		return false, nil
+	}
+	deadline, err := time.Parse(time.RFC3339, t)
+	if err != nil {
+		return false, fmt.Errorf("Managed Decommission annotation must be a valid RFC3339 timestamp: %w", err)
+	}
+	return deadline.After(time.Now()), nil
+}
+
```

Comment:
Can you change the commit description? The `rollingUpdate` description have a lot of problems.